task_manager_using_typescript/src/data_file.ts

```diff
@@ -0,0 +1,32 @@
+import { InterfaceTask } from "./interfaces";
+
+let baseUrl: string = "https://66a0bb4b7053166bcabc8698.mockapi.io/Tasklist";
+
+export const fetchData = <T>(
+  url: string = "",
+  option: RequestInit = {}
+): Promise<T> => {
+  const fullUrl = `${baseUrl}${url}`;
+
+  return fetch(fullUrl, option)
```

Comment:
What if the response is ok, with some error will it goes to handle by catch

Reply:
If data is  encrypted then promise is rejected and it goes to catch block.